calcom/cal.com
# fix: Add `refresh_token` placeholder when parsing refresh token responses

## What does this PR do?



When fetching new access tokens from OAuth apps not every response included a new refresh token. As a part of credential syncing a refresh token placeholder was added assuming that normal access token responses would already contain a refresh token. This PR only adds the placeholder if credential syncing is enabled.

This issue was first noticed in Outlook but should apply to other apps.

Fixes # (issue)



## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change





## How should this be tested?



- Install Outlook
- In the DB set the expiry date to in the past to force a refresh
- In the token response ensure that a refresh token was not passed
- The refresh token placeholder should not have been added if a refresh token was not present in the MS token response

## Mandatory Tasks


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
